frontend/src/components/HomePage.vue

```diff
@@ -144,14 +141,16 @@ export default defineComponent({
       this.modalObjectType = type
       this.modalObject = object
       this.modalMode = mode
-      if (this.modal) {
-        this.modal.show()
+      if ((this.$refs.modalComp as typeof ModalComponent).modal) {
+        ;(this.$refs.modalComp as typeof ModalComponent).modal.show()
       }
     },
     hideModal() {
-      if (this.modal) {
-        this.modal.hide()
+      if ((this.$refs.modalComp as typeof ModalComponent).modal) {
+        ;(this.$refs.modalComp as typeof ModalComponent).hideModal()
       }
+    },
+    reset() {
```

Comment:
Vielleicht kannst du der Methode in auf den verschiedenen Pages einen eindeutigeren Namen geben, weil die Page wird ja nicht reset. Eher z.B. 'resetForms'